python/imageutils.py

```diff
@@ -111,19 +126,20 @@ def reconstruct_block_block(self, legend):
 
         NOTE: currently only supports nifti blocks as it uses 'bitpix' to determine number of bytes per voxel. Element is specific to nifti headers
         """
-
         
-        rec_header = self.proxy.header
-        bytes_per_voxel = rec_header['bitpix']/8 
-        
-        rec_dims = self.proxy.header.get_data_shape()
-     
-        y_size = rec_dims[0]
-        z_size = rec_dims[1]
-        x_size = rec_dims[2]
-
-        #print self.filepath
-        with open(self.filepath, "r+b") as reconstructed:
+
+        with open(self.filepath, self.file_access()) as reconstructed:
+
+            if self.proxy is None:
```

Comment:
yeah, i just wanted to avoid opening the file twice when it only needs to be done once. I can create a function called reconstruct_img which writes header to file and passes the file object to either reconstruct_block_block, reconstruct_block_slice or reconstruct_slice_slice